pandas/io/json/json.py

```diff
@@ -349,22 +382,71 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
         lines = list(StringIO(json.strip()))
         json = '[' + ','.join(lines) + ']'
 
+    return _get_obj(json, **kwargs)
+
+
+def _get_obj(json, **kwargs):
+    typ = kwargs['typ']
+    dtype = kwargs['dtype']
+    kwargs = {k: v for k, v in kwargs.items() if k != 'typ'}
     obj = None
     if typ == 'frame':
-        obj = FrameParser(json, orient, dtype, convert_axes, convert_dates,
-                          keep_default_dates, numpy, precise_float,
-                          date_unit).parse()
+        obj = FrameParser(json, **kwargs).parse()
 
     if typ == 'series' or obj is None:
         if not isinstance(dtype, bool):
             dtype = dict(data=dtype)
-        obj = SeriesParser(json, orient, dtype, convert_axes, convert_dates,
-                           keep_default_dates, numpy, precise_float,
-                           date_unit).parse()
+        obj = SeriesParser(json, **kwargs).parse()
 
     return obj
 
 
+class JsonLineReader(BaseIterator):
+    """
+    Iterates over a JSON document that is formatted with one JSON record per
+    line. The `chunksize` initialization parameter controls how many lines are
+    read per iteration.
+
+    We explicitly override the index on the return value so that the index of
+    the resulting object will be like `range(len(obj))`. If we didn't do this,
+    it would have index like `range(chunksize) * number_chunks.`
+    This is so that `read_json(lines=True)` will return an identical object to
+    `read_json(lines=True, chunksize=n)`.
+    """
+    def __init__(self, filepath_or_buffer, chunksize, encoding, **kwargs):
+
+        try:
+            self.iterator, _ = _get_handle(filepath_or_buffer, 'r',
+                                           encoding=encoding)
+        except:
+            if hasattr(filepath_or_buffer, 'read'):
+                self.iterator = filepath_or_buffer
+            else:
+                raise ValueError("cannot read json from given input")
+        self.chunksize = chunksize
+        self.kwargs = kwargs
+        self.nrows_seen = 0
+
+    def __next__(self):
+        lines = list(islice(self.iterator, self.chunksize))
```

Comment:
is lines possible to be empty?

Reply:
Yes, it's empty after all the lines have been read, which triggers the StopIteration.